app/lib/rocketchat/methods/search.ts

```diff
@@ -0,0 +1,98 @@
+import { Q } from '@nozbe/watermelondb';
+
+import { sanitizeLikeString } from '../../database/utils';
+import database from '../../database/index';
+import { spotlight } from '../services/restApi';
+import isGroupChat from './isGroupChat';
+import { ISearch, ISearchLocal, SubscriptionType } from '../../../definitions';
+
+let debounce: null | ((reason: string) => void) = null;
+
+export const search = async ({ text = '', filterUsers = true, filterRooms = true }): Promise<(ISearch | ISearchLocal)[]> => {
+	const searchText = text.trim();
+
+	if (debounce) {
+		debounce('cancel');
+	}
+
+	const localSearchData = await localSearch({ text, filterUsers, filterRooms });
+	const usernames = localSearchData.map(sub => sub.name);
+
+	const data: (ISearch | ISearchLocal)[] = localSearchData;
+
+	try {
+		if (localSearchData.length < 7) {
+			const { users, rooms } = (await Promise.race([
+				spotlight(searchText, usernames, { users: filterUsers, rooms: filterRooms }),
+				new Promise((resolve, reject) => (debounce = reject))
+			])) as { users: ISearch[]; rooms: ISearch[] };
+
+			if (filterUsers) {
+				users
+					.filter((item1, index) => users.findIndex(item2 => item2._id === item1._id) === index) // Remove duplicated data from response
+					.filter(user => !data.some(sub => user.username === sub.name)) // Make sure to remove users already on local database
+					.forEach(user => {
+						data.push({
+							...user,
+							rid: user.username,
+							name: user.username,
+							t: SubscriptionType.R_THREAD,
```

Comment:
This is actually `'d'`, so you should use SubscriptionType.DIRECT.

We're just mapping the users returned from the search and adding the type manually to make things easier.